backend/enums/activity.js

```diff
@@ -71,6 +71,26 @@ export default class Activity {
     return activity
   }
 
+  static async retrieveInterests(userId) {
+    const interests = []
+    const unconvertedInterests = await this.getPostsForInterests(userId)
+    await Promise.all(
+      unconvertedInterests.map(async (activity) => {
+        const post = await this.convertToFeedActivity(
+          this.SHARED,
+          activity.postId,
+          activity.userId,
+          activity.createdAt
+        )
+        interests.push(post)
+      })
+    )
+
+    interests.sort((a, b) => (a.timestamp < b.timestamp ? 1 : -1))
```

Comment:
```suggestion
    interests.sort((a, b) => (b.timestamp - a.timestamp))
```

I don't think you take into account equal timestamps. Perhaps this is a concise way to do it?